File: QA.Engine.Administration.WebApp.Core/Models/RegionViewModel.cs
using System.Collections.Generic;
using System.Linq;
using QA.Engine.Administration.WebApp.Core.Annotations;

namespace QA.Engine.Administration.WebApp.Core.Models
{
    [TypeScriptType]
    public class RegionViewModel
    {
        public int Id { get; set; }
        public string Alias { get; set; }
        public int? ParentId { get; set; }
        public string Title { get; set; }
        public List<RegionViewModel> Children { get; set; }
        public bool HasChildren { get { return Children?.Any() ?? false; } }
    }
}

File: QA.Engine.Administration.WebApp.Core/ClientApp/models/RegionViewModel.ts

module Models {

    // $Classes/Enums/Interfaces(filter)[template][separator]
    // filter (optional): Matches the name or full name of the current item. * = match any, wrap in [] to match attributes or prefix with : to match interfaces or base classes.
    // template: The template to repeat for each matched item
    // separator (optional): A separator template that is placed between all templates e.g. $Properties[public $name: $Type][, ]
    
    // More info: http://frhagn.github.io/Typewriter/
    
    
    export class RegionViewModel {
        
        // ID
        public id: number = 0;
        // ALIAS
        public alias: string = null;
        // PARENTID
        public parentId: number = null;
        // TITLE
        public title: string = null;
        // CHILDREN
        public children: RegionViewModel[] = [];
        // HASCHILDREN
        public hasChildren: boolean = false;
    }
}
